dashboard/app/controllers/activities_controller.rb

```diff
@@ -113,29 +160,60 @@ def milestone_logger
     @@milestone_logger ||= Logger.new("#{Rails.root}/log/milestone.log")
   end
 
-  def track_progress_for_user
-    authorize! :create, Activity
-    authorize! :create, UserLevel
+  def track_level_solution(level:, program:, image:)
```

Comment:
Will the use of required keyword arguments in this PR require Ruby >= 2.1? If so, then PR will be blocked on finishing (and enforcing) the 2.2 upgrade, including making sure everyone has upgraded their workstations first.